messages_format.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+message RequestMsg {
+// control data
+  string method = 1;
+  string scheme = 2;
+  string authority = 3;
+  string path = 4;
+// 
+  string[] header_fields = 5;
```

Comment:
but maybe map<string, string> is a better option?

Reply:
That would be easier to decode on the server-side, yeah.

Reply:
Yup. One concern from past experience with protobuf: maps slow down encoding/decoding up to several times. But probably when we get to these kind of issues with protobuf, we'll switch to BHTTP :-)